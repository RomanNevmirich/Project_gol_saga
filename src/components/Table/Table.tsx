import React from "react";
import { Cell } from '../Cell';

export type TableType = (0|1)[][];

export type TableObject = {
    table: TableType;
};

export const Table = ( { table }: TableObject ) => {
  //console.log(table);
  return <div style={{display: 'grid', gridTemplateColumns: `repeat(${table[0]?.length ?? 0}, 12px)`}}>
    {table.map((rows, i) => rows.map(
      (col, j) => 
        <Cell key={`${i}|${j}`} value={table[i][j]}/>
      )
    )}
  </div>
};