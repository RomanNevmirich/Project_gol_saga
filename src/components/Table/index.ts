export { Table } from './Table';
export type { TableType } from './Table';