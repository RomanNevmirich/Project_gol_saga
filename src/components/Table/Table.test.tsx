import React from 'react';
import { render, screen, fireEvent, createEvent } from '@testing-library/react';
import { Table, TableType, TableObject } from './Table';

describe('Table component check', () => {
  const table: TableObject = {table: [[0,0,0,0],
                                      [0,1,1,0],
                                      [0,1,1,0],
                                      [0,0,0,0]]};

  it('Table render fill', () => {
    const { asFragment } = render(<Table {...table} />);
    expect(asFragment()).toMatchSnapshot();
  });

});