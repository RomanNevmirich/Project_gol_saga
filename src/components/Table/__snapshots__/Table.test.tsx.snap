// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component check Table render empty (?) 1`] = `
<DocumentFragment>
  <div
    style="display: grid; grid-template-columns: repeat(0, 12px);"
  />
</DocumentFragment>
`;

exports[`Table component check Table render fill 1`] = `
<DocumentFragment>
  <div
    style="display: grid; grid-template-columns: repeat(4, 12px);"
  >
    <div
      class="sc-bczRLJ hcpULb"
      value="0"
    />
    <div
      class="sc-bczRLJ hcpULb"
      value="0"
    />
    <div
      class="sc-bczRLJ hcpULb"
      value="0"
    />
    <div
      class="sc-bczRLJ hcpULb"
      value="0"
    />
    <div
      class="sc-bczRLJ hcpULb"
      value="0"
    />
    <div
      class="sc-bczRLJ klZATh"
      value="1"
    />
    <div
      class="sc-bczRLJ klZATh"
      value="1"
    />
    <div
      class="sc-bczRLJ hcpULb"
      value="0"
    />
    <div
      class="sc-bczRLJ hcpULb"
      value="0"
    />
    <div
      class="sc-bczRLJ klZATh"
      value="1"
    />
    <div
      class="sc-bczRLJ klZATh"
      value="1"
    />
    <div
      class="sc-bczRLJ hcpULb"
      value="0"
    />
    <div
      class="sc-bczRLJ hcpULb"
      value="0"
    />
    <div
      class="sc-bczRLJ hcpULb"
      value="0"
    />
    <div
      class="sc-bczRLJ hcpULb"
      value="0"
    />
    <div
      class="sc-bczRLJ hcpULb"
      value="0"
    />
  </div>
</DocumentFragment>
`;
